pokedex_async_redux/lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -1,68 +1,141 @@
-      body: pokemons.when(
-        (data) {
-          return GridView.builder(
-            itemCount: data.length,
-            gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 2),
-            itemBuilder: (_, index) {
-              final pokemon = data[index];
-              return PokemonCard(
-                pokemon: pokemon,
-                onTap: () => _navigateToPokemonDetailsPage(context: context, pokemon: pokemon),
-              );
+      body: Column(
+        children: [
+          PokemonSearchBar(
+            textController: _searchController,
+            clearSearchBar: _clearSearchedPokemons,
+            textChanged: (val) {
+              setState(() {
+                _searchQuery = val;
+              });
+              _duringPokemonSearch(val);
             },
-          );
-        },
-        loading: () => Container(
-          color: mainBgColor,
-          child: Center(
-            child: Image.asset(
-              pikachuRunningImage,
-              height: 210,
-              width: 210,
-            ),
           ),
-        ),
-        error: (errorMessage) {
-          WidgetsBinding.instance.addPostFrameCallback(
-            (_) => _showErrorMessageSnackbar(context, errorMessage),
-          );
-          return Center(
-            child: Column(
-              mainAxisAlignment: MainAxisAlignment.center,
-              children: [
-                const Text(noPokemonsAvailableLabel),
-                const VerticalSpace(height: 20),
-                Image.asset(
-                  pokedexErrorImage,
-                  height: 120,
-                  width: 120,
+          Expanded(
+            child: widget.pokemons.when(
+              (data) {
+                final filteredPokemons =
+                    data.where((pokemon) => pokemon.name.toLowerCase().contains(_searchQuery.toLowerCase()));
+                if (filteredPokemons.isEmpty) {
+                  return Center(
+                    child: Column(
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children: [
+                        const Text(
+                          errorSearchMessage,
+                          style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
+                        ),
+                        Image.asset(
+                          magikarpFloppingImage,
+                          height: pokemonImageSize,
+                          width: pokemonImageSize,
+                        ),
+                      ],
+                    ),
+                  );
+                } else {
+                  return GridView.builder(
+                    itemCount: filteredPokemons.length,
+                    gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 2),
+                    itemBuilder: (_, index) {
+                      final pokemon = filteredPokemons.elementAt(index);
```

Comment:
how about if you create a validation for something like this here?

`final pokemon = isSearching ? searchedPokemons : pokemons;`

same goes for your empty label message:

`final sampleLabelMessage = isSearching ? noSearchedPokemonsLabel : noPokemonsRetrievedLabel`